Modules/Microsoft365DSC/DSCResources/MSFT_TeamsGroupPolicyAssignment/MSFT_TeamsGroupPolicyAssignment.psm1

```diff
@@ -0,0 +1,439 @@
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $CurrentValues = Get-TargetResource @PSBoundParameters
+
+    Write-Verbose -Message "Current Values: $(Convert-M365DscHashtableToString -Hashtable $CurrentValues)"
+    Write-Verbose -Message "Target Values: $(Convert-M365DscHashtableToString -Hashtable $PSBoundParameters)"
+
+    $TestResult = Test-M365DSCParameterState -CurrentValues $CurrentValues `
+        -Source $($MyInvocation.MyCommand.Source) `
+        -DesiredValues $PSBoundParameters `
+        -ValuesToCheck @('Ensure', `
+            'GroupDisplayname', `
+            'PolicyType', `
+            'PolicyName', `
+            'Priority')
+
+    Write-Verbose -Message "Test-TargetResource returned $TestResult"
+
+    return $TestResult
+}
+
+function Export-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.String])]
+    param
+    (
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint
+    )
+    $InformationPreference = 'Continue'
+    $ConnectionMode = New-M365DSCConnection -Workload 'MicrosoftTeams' -InboundParameters $PSBoundParameters
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName -replace 'MSFT_', ''
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    try
+    {
+        [array]$instances = Get-CsGroupPolicyAssignment
+        if ($instances.Length -eq 0)
+        {
+            Write-Host $Global:M365DSCEmojiGreenCheckMark
+        }
+        else
+        {
+            Write-Host "`r`n" -NoNewline
+        }
+        $dscContent = [System.Text.StringBuilder]::new()
+        $j = 1
+        $totalCount = $instances.Length
+        foreach ($item in $instances)
+        {
+            $Group = Find-CsGroup -SearchQuery $item.GroupId
+            if ($null -eq $totalCount)
+            {
+                $totalCount = 1
+            }
+            Write-Host "    > [$j/$totalCount] GroupPolicyAssignment {$($Group.DisplayName)}" -NoNewline
+            $results = @{
+                GroupDisplayname      = $Group.DisplayName
+                GroupId               = $Group.Id
+                PolicyType            = $item.PolicyType
+                PolicyName            = $item.PolicyName
+                Priority              = $item.Priority
+                Ensure                = 'Present'
+                Id                    = $j
```

Comment:
Refer to comment above. Id shouldn't be a property of this resource if it's not something that exists on the M365 server side.